src/tests/performance.rs

```diff
@@ -0,0 +1,234 @@
+    //                  ╠════╬════════╬══════════╬═══════════╬═════════════╬═════════════╬════════╣
+    //                  ║ 50 ║    100 ║       42 ║        58 ║           0 ║          42 ║  3.73s ║
+    //                  ║ 50 ║  1_000 ║       42 ║       958 ║           0 ║          42 ║  2.33s ║
+    //                  ║ 50 ║  5_000 ║       42 ║     4_958 ║           0 ║          42 ║  2.83s ║
+    //                  ║ 50 ║ 10_000 ║       42 ║     9_958 ║           0 ║          42 ║  3.67s ║
+    //                  ║ 50 ║ 15_000 ║       42 ║    14_958 ║           0 ║          42 ║  6.19s ║
+    //                  ║ 50 ║ 20_000 ║       42 ║    19_958 ║           0 ║          42 ║ 20.13s ║
+    //                  ╚════╩════════╩══════════╩═══════════╩═════════════╩═════════════╩════════╝
+    //
+    // Zebra: Ignores the set limit, and does not appear to have an actual limit
+    //        set. Example runs (with N=50):
+    //
+    //                  ╔════╦════════╦══════════╦═══════════╦═════════════╦═════════════╦════════╗
+    //                  ║ N  ║   M    ║ Accepted ║ Rejected  ║ Terminated  ║ Max active  ║  Time  ║
+    //                  ╠════╬════════╬══════════╬═══════════╬═════════════╬═════════════╬════════╣
+    //                  ║ 50 ║    100 ║      100 ║         0 ║           0 ║         100 ║  0.59s ║
+    //                  ║ 50 ║  1_000 ║    1_000 ║         0 ║           0 ║       1_000 ║  1.17s ║
+    //                  ║ 50 ║  5_000 ║    4_962 ║        38 ║           0 ║       4_962 ║  3.55s ║
+    //                  ║ 50 ║ 10_000 ║    9_777 ║       223 ║           0 ║       9_777 ║  8.96s ║
+    //                  ║ 50 ║ 15_000 ║   13_782 ║     1_218 ║      12_077 ║      12_651 ║ 66.57s ║
+    //                  ║ 50 ║ 20_000 ║    4_255 ║    15_745 ║           1 ║       4_255 ║ 34.06s ║
+    //                  ╚════╩════════╩══════════╩═══════════╩═════════════╩═════════════╩════════╝
+    //
+    // The example runs are pretty representative. In particular, zebra starts exhibiting weird
+    // behaviour at 15_000.
+
+    /// maximum peers to configure node with
+    const MAX_PEERS: usize = 50;
+    /// number of test peer nodes to spin-up
+    const TEST_PEER_COUNT: usize = 100;
+
+    // channel for peer event management (ensure buffer is more than large enough)
+    let (event_tx, mut event_rx) = tokio::sync::mpsc::channel::<PeerEvent>(TEST_PEER_COUNT * 3);
+
+    // start node
+    let mut node: Node = Default::default();
+    node.initial_action(Action::WaitForConnection(new_local_addr()))
+        .max_peers(MAX_PEERS)
+        .start()
+        .await;
+
+    // start peer event manager
+    let event_manager = tokio::spawn(async move {
+        let mut stats = ConnectionStats::default();
+
+        loop {
+            match event_rx.recv().await.unwrap() {
+                PeerEvent::Rejected => stats.rejected += 1,
```

Comment:
Note for the future: if we want prometheus support, we'll likely want to replace these with https://docs.rs/metrics/0.16.0/metrics/.